pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -278,10 +278,22 @@ def model_to_device(self):
             torch.cuda.set_device(self.root_device)
         self.model.to(self.root_device)
 
-    def reduce(self, output, group: Optional[Any] = None, reduce_op: Optional[Union[ReduceOp, str]] = None):
-        if isinstance(output, torch.Tensor):
-            output = sync_ddp_if_available(output, group, reduce_op)
-        return output
+    def reduce(self, tensor, group: Optional[Any] = None, reduce_op: Optional[Union[ReduceOp, str]] = "mean"):
+        """
+        Reduces a tensor from several distributed processes to one aggregated tensor.
+
+        Args:
+            tensor: the tensor to sync and reduce
+            group: the process group to gather results from. Defaults to all processes (world)
+            reduce_op: the reduction operation. Defaults to 'mean'/'avg'.
+                Can also be a string 'sum' to calculate the sum during reduction.
+
+        Return:
+            reduced value, except when the input was not a tensor the output remains is unchanged
+        """
+        if isinstance(tensor, torch.Tensor):
+            tensor = sync_ddp_if_available(tensor, group, reduce_op=(reduce_op or "mean"))
```

Comment:
Changed this, to make the default "mean". 
Let me know what you think @tchaton @justusschock 

If you agree with this, I will change the PR title, otherwise I will revert back to a pure docs update.

Reply:
I think it makes sense to have mean as default. What is PyTorch default ?

Reply:
pytorch differentiates between reduce (only process 0 gets result) and all_reduce (all processes get the result). The default reduction is SUM. 
https://pytorch.org/docs/stable/distributed.html?highlight=all_reduce#torch.distributed.all_reduce

Our internal implementation uses `torch.distributed.all_reduce` with SUM and for mean we divide by world_size manually.